packages/yarnpkg-core/sources/Cache.ts

```diff
@@ -244,7 +246,11 @@ export class Cache {
 
       await xfs.chmodPromise(refetchPath, 0o644);
 
-      return await validateFile(cachePath, refetchPath);
+      const result = await validateFile(cachePath, refetchPath);
+      if (!result[0])
+        throw new Error(`Not expected`);
```

Comment:
when `refetchPath` is provided, it's always checksum is always valid. 
Basically it's safe guard, to avoid bugs on future changes. 
Can be removed. Or changed to "Invariant, should always be true"

Reply:
I see, should be something like https://github.com/yarnpkg/berry/blob/fc57d688a2fd48eebbdeff75f69d4d5ca2b1e567/packages/yarnpkg-core/sources/Cache.ts#L275, for example
```
Assertion failed: Expected a valid checksum
```

Reply:
fixed